src/reblox/grid.rs

```diff
@@ -0,0 +1,595 @@
+                BlockType::S => [pos, pos + 1, pos - grid_width, pos - grid_width - 1],
+                BlockType::T => [pos - 1, pos, pos + 1, pos - grid_width],
+                BlockType::Z => [pos - 1, pos, pos - grid_width, pos - grid_width + 1],
+                _ => [0, 0, 0, 0],
+            },
+        1 =>
+            match block_type {
+                BlockType::I => [pos + 1, pos, pos - 1, pos - 2],
+                BlockType::J => [pos + 1, pos, pos - 1, pos + grid_width - 1],
+                BlockType::L => [pos + 1, pos, pos - 1, pos - grid_width - 1],
+                BlockType::O => [pos, pos - 1, pos - grid_width, pos - grid_width - 1],
+                BlockType::S => [pos - grid_width, pos, pos - 1, pos + grid_width - 1],
+                BlockType::T => [pos + grid_width, pos, pos - grid_width, pos - 1],
+                BlockType::Z => [pos + grid_width, pos, pos - 1, pos - grid_width - 1],
+                _ => [0, 0, 0, 0],
+            },
+        2 =>
+            match block_type {
+                BlockType::I => [pos - grid_width, pos, pos + grid_width, pos - grid_width * 2],
+                BlockType::J => [pos - grid_width, pos, pos + grid_width, pos + grid_width + 1],
+                BlockType::L => [pos - grid_width, pos, pos + grid_width, pos + grid_width - 1],
+                BlockType::O => [pos, pos - 1, pos + grid_width, pos + grid_width - 1],
+                BlockType::S => [pos - 1, pos, pos + grid_width, pos + grid_width + 1],
+                BlockType::T => [pos - 1, pos, pos + 1, pos + grid_width],
+                BlockType::Z => [pos + 1, pos, pos + grid_width, pos + grid_width - 1],
+                _ => [0, 0, 0, 0],
+            },
+        3 =>
+            match block_type {
+                BlockType::I => [pos + 1, pos, pos - 1, pos - 2],
+                BlockType::J => [pos - 1, pos, pos + 1, pos - grid_width + 1],
+                BlockType::L => [pos - 1, pos, pos + 1, pos + grid_width + 1],
+                BlockType::O => [pos, pos + 1, pos + grid_width, pos + grid_width + 1],
+                BlockType::S => [pos + grid_width, pos, pos + 1, pos - grid_width + 1],
+                BlockType::T => [pos + 1, pos + grid_width, pos, pos - grid_width],
+                BlockType::Z => [pos - grid_width, pos, pos + 1, pos + grid_width + 1],
+                _ => [0, 0, 0, 0],
+            },
+        _ => panic!("fn pr2piece: Invalid rotation value!"),
+    }
+}
+
+/// (0, 0) is bottom left
+pub struct Grid {
+    pub grid: [BlockType; (GRID_WIDTH * (GRID_HEIGHT + 1)) as usize],
+    pub grid_size: u8,
+    falling_type: BlockType,
+    falling_pos: u8,
+    falling_rot: u8,
+    next_falling_type: BlockType,
+    next_falling_rot: u8,
+    pub rng: PseudoRandom,
+    level: u8,
+    lines_cleared: u32,
+    elapsed_time: Duration,
+    cached_fall_rate: Duration,
+    pub dead: bool,
+}
+
+impl Grid {
+    pub fn new() -> Grid {
+        Grid {
```

Comment:
This struct is like 90% `Default::default()`. Derive `Default` and simply override the respecitve field through the `.. Grid::default()` syntax.